code/modules/psionics/interface/ui_hub.dm

```diff
@@ -91,6 +91,10 @@
 			var/psionic_path = text2path(params["buy"])
 			if(ispath(psionic_path))
 				var/singleton/psionic_power/P = GET_SINGLETON(psionic_path)
-				P.apply(owner)
-				to_chat(owner, SPAN_NOTICE("You are now capable of using [P.name]."))
-				return TRUE
+				if(P.point_cost > owner.psi.psi_points)
+					to_chat(owner, SPAN_WARNING("You can't afford that!"))
+					return
+				if(P.apply(owner))
+					to_chat(owner, SPAN_NOTICE("You are now capable of using [P.name]."))
+					owner.psi.psi_points = max(owner.psi.psi_points - P.point_cost, 0)
+					return TRUE
```

Comment:
```suggestion
				if(P.apply(owner))
					to_chat(owner, SPAN_NOTICE("You are now capable of using [P.name]."))
					log_and_message_admins("[key_name(owner)] bought ability [P.name] for [P.point_cost] points.", owner, get_turf(owner))
					owner.psi.psi_points = max(owner.psi.psi_points - P.point_cost, 0)
					return TRUE
```

Might be nice to also get this into our admin logs, since buying items using an uplink is also logged. We can just VV the people in question within a round and look at abilities, but after a round ended there's no trace what abilities they bought.